classes/Validate.php

```diff
@@ -504,11 +509,15 @@ public static function isCleanHtml($html, $allow_iframe = false)
      *
      * @param string $reference Product reference to validate
      *
+     * product reference and manufacturer reference are max 64 characters in db
+     *
+     * This should really be handled in the DAL layer and it should throw an error which can then be used elsewhere
+     *
      * @return bool Validity is ok or not
      */
     public static function isReference($reference)
     {
-        return preg_match(Tools::cleanNonUnicodeSupport('/^[^<>;={}]*$/u'), $reference);
+        return Tools::strlen($reference) <= 64;
```

Comment:
I'm never comfortable with value in code, especially since is duplicated in many place. Could you create a const that you will be able to re-use anywhere you want ?